packages/core/integration-tests/test/html.js

```diff
@@ -273,8 +273,9 @@ describe('html', function() {
       path.join(distDir, 'index.html'),
       'utf8',
     );
+
     assert(
-      /<\/script>\s*<link rel="stylesheet" href="[/\\]{1}html-css-optional-elements\.[a-f0-9]+\.css"><h1>Hello/.test(
+      /^<link rel="stylesheet" href="[/\\]html-css-optional-elements\.[a-f0-9]+\.css">\s*<script src="[/\\]other\.[a-f0-9]+\.js"><\/script>\s*<h1>Hello/m.test(
```

Comment:
My change is to insert the sibling before the first non-metadata tag or script.

So previously

```html
<script src="/other.7576ab85.js"></script> <!-- entry1-->
<link rel="stylesheet" href="/html-css-optional-elements.03af036d.css"> <!-- inserted -->
<h1>Hello world</h1>
<script src="/html-css-optional-elements.9d40fb57.js"></script> <!-- entry2 -->
```

Now:

```html
<link rel="stylesheet" href="/html-css-optional-elements.03af036d.css">  <!-- inserted -->
<script src="/other.7576ab85.js"></script> <!-- entry1 -->
<h1>Hello world</h1>
<script src="/html-css-optional-elements.9d40fb57.js"></script> <!-- entry2 -->
```

I guess this is fine?
To keep the old behaviour we would need to have a different logic when a link (order doesn't matter) or a script (order is important) is inserted.